### ISSUE TITLE
Reports profiles don’t output to the correct Desktop folder if OneDrive is in use

### ISSUE THREAD

**Issue by YenShihHung** [alterNERDtive/bindED#43]
Because of the Windows 11 system directory change, the original output path is no longer applicable.

**Comment by alterNERDtive** [comment]
And what would that directory change be? I can fix it if I know, but I don't have a Windows PC.

**Comment by YenShihHung** [comment]
To be compatible with different versions of operating systems, it is recommended to grab the directory location from the registry. Windows 11’s OneDrive changes the original desktop directory

[https://learn.microsoft.com/en-us/dotnet/api/system.environment.getfolderpath?view=net-9.0](url)

[https://support.microsoft.com/en-us/topic/how-to-redirect-user-shell-folders-to-a-specified-path-by-using-profile-maker-ed6289ae-1f9c-b874-4e8c-20d23ea65b2e](url)

**Comment by alterNERDtive** [comment]
Yeah, I’m not going to look into how to grab this from the registry, nor am I going to support OneDrive in any way.

If someone else wants to tackle it: pull requests welcome. Potentially also one that changes the default output folder to something else instead of the Desktop.

**Comment by alterNERDtive** [comment]
In the meantime, if you need the reports to work, you can just create `C:\users\<user>\Desktop` manually.

**Comment by YenShihHung** [comment]
Yeah, that's how I handle it too

**Comment by alterNERDtive** [comment]
@YenShihHung I feel like I should elaborate: Just because _I_ personally am not going to fix it, I would still consider this a bug. So I’ll keep it open for people that have the same problem and/or want to contribute a fix.

**Comment by alterNERDtive** [comment]
Heh, just realized that I actually _cannot_ really fix this; it’s part of the profiles and not the plugin, and I don’t have a Windows machine to run VA on :)
